pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -943,14 +1081,18 @@ func (c *Controller) AdditionalPodSubscriptions(
 	// The client wouldn't be explicitly subscribed to Pod1, so it would normally ignore it.
 	// Since it is a part of VIP1 which we are subscribe to, add it to the subscriptions
 	for s := range allAddresses {
-		for _, wl := range c.ambientIndex.Lookup(s.Name) {
+		cNetwork := c.Network(s.Name, make(labels.Instance, 0)).String()
+		for _, wl := range c.ambientIndex.Lookup(cNetwork + "/" + s.Name) {
 			// We may have gotten an update for Pod, but are subscribe to a Service.
 			// We need to force a subscription on the Pod as well
-			for addr := range wl.VirtualIps {
-				t := types.NamespacedName{Name: addr}
-				if currentSubs.Contains(t) {
-					shouldSubscribe.Insert(types.NamespacedName{Name: wl.ResourceName()})
-					break
+			switch addr := wl.Address.Type.(type) {
+			case *workloadapi.Address_Workload:
```

Comment:
let's add a case to explicitly ignore the service with the reason why so it doesn't look like it was just forgotten

Reply:
I had actually forgotten to handle Service in these two spots.  here, I felt the Service object should be properly handled and subscribed to on a match